### DISCUSSION TITLE
Bug: discussion category is not pre-filled after visiting link on category_choices page

### DISCUSSION THREAD

**Discussion by roryokane** [community/community#3175] [Discussions]
## Background/setup

In the [Snowpack project’s new issue template](https://github.com/snowpackjs/snowpack/issues/new), there are instructions with a link to create a new discussion:

> ----   📝 Create New:        https://github.com/snowpackjs/snowpack/discussions/category_choices

On that [category_choices page](https://github.com/snowpackjs/snowpack/discussions/category_choices), I see five categories. Let’s pick the General category as an example. I click the “Start discussion” button next to it, taking me to https://github.com/snowpackjs/snowpack/discussions/new?category_id=16417347.

## Actual behavior

Now I am on the “Start a new discussion” page. Yet the menu for the discussion category just says “Select Category”. I have to click to open the menu and then click the General category for the choice to actually be made. It seems the `category_id=16417347` URL parameter added by the category_choices page’s link had no effect.

## Expected behavior

I would expect, given my selection of General on the previous page, that the discussion category menu would default to General.

**Comment by simonswiss** [comment]
Confirming this on the Tailwind Labs discussions: https://github.com/tailwindlabs/tailwindcss/discussions

`category_id` has no effect, so for example the following line will **not** have the `help` category preselected:

https://github.com/tailwindlabs/tailwindcss/discussions/new?category_id=65334

However, the `category` query param works, so this will actually work with the category preselected properly:

https://github.com/tailwindlabs/tailwindcss/discussions/new?category=help

I imagine the issue with this approach is whenever you change a category name, you may break a lot of links 😅
